agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -574,7 +574,7 @@ func (dg *dockerGoClient) createContainer(ctx context.Context,
 		return DockerContainerMetadata{Error: CannotGetDockerClientError{version: dg.version, err: err}}
 	}
 
-	dockerContainer, err := client.ContainerCreate(ctx, config, hostConfig, &network.NetworkingConfig{}, name)
+	dockerContainer, err := client.ContainerCreate(ctx, config, hostConfig, &network.NetworkingConfig{}, nil, name)
```

Comment:
Passing nil to ContainerCreate means that the client library will ignore
this arg and not pass it in the list of arguments it submits to the
docker API.

see https://github.com/moby/moby/blob/4c0d75bc8e8311407040ae7c930f5a2c2c9c03aa/client/container_create.go#L49

We don't want to pass the platform arg because we are not using it and
this is the simplest way to ensure continued support of older docker
versions.